pkg/collector/corechecks/cluster/kubernetes_state_transformers.go

```diff
@@ -42,6 +43,47 @@ var (
 	}
 )
 
+// podPhaseTransformer sends status phase metrics for pods, the tag phase has the pod status
+func podPhaseTransformer(s aggregator.Sender, name string, metric ksmstore.DDMetric, tags []string) {
+	if metric.Val != 1.0 {
+		// Only consider active metrics
+		return
+	}
+	// As opposed to KSM check v1, send only the active phase and keep the pod label
+	s.Gauge(ksmMetricPrefix+"pod.status_phase", 1, "", tags)
+}
+
+var allowedWaitingReasons = map[string]struct{}{
+	"errimagepull":      {},
+	"imagepullbackoff":  {},
+	"crashloopbackoff":  {},
+	"containercreating": {},
+}
+
+// containerWaitingReasonTransformer validates the container waiting reasons for metric kube_pod_container_status_waiting_reason
+func containerWaitingReasonTransformer(s aggregator.Sender, name string, metric ksmstore.DDMetric, tags []string) {
+	if reason, found := metric.Labels["reason"]; found {
+		if _, allowed := allowedWaitingReasons[strings.ToLower(reason)]; allowed {
```

Comment:
Why are we keeping only some reasons? We already check if `reason` label is present, why not trust KSM and report as-is?

Reply:
I believe it's a protection against high cardinality, see https://github.com/DataDog/integrations-core/blob/master/kubernetes_state/datadog_checks/kubernetes_state/kubernetes_state.py#L589 and https://github.com/DataDog/integrations-core/pull/1884

Reply:
We should document it clearly in the code then

Reply:
yes good point, done 👌